### ISSUE TITLE
a11y: Mark nodes as hidden when they're non-rendered

### ISSUE THREAD

**Issue by alice** [servo/servo#46809]
If a DOM node is not rendered, its corresponding accessibility node should be marked as `hidden`.

This state, like `display: none` and `aria-hidden: true`, applies to all the node's children as well, without needing to set the state separately for each one.
